<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>797</width>
    <height>198</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTableWidget" name="tableWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>781</width>
      <height>45</height>
     </rect>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="editTriggers">
     <set>QAbstractItemView::NoEditTriggers</set>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>100</number>
    </attribute>
    <row>
     <property name="text">
      <string>Deque:</string>
     </property>
    </row>
   </widget>
   <widget class="QPushButton" name="pushBackButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>88</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>pushBack</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushFrontButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>88</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>pushFront</string>
    </property>
   </widget>
   <widget class="QPushButton" name="popBackButton">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>70</y>
      <width>88</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>popBack</string>
    </property>
   </widget>
   <widget class="QPushButton" name="popFrontButton">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>110</y>
      <width>88</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>popFront</string>
    </property>
   </widget>
   <widget class="QPushButton" name="swapButton">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>70</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>swapMinMax</string>
    </property>
   </widget>
   <widget class="QPushButton" name="isEmptyButton">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>70</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>isEmpty</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearButton">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>110</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>clear</string>
    </property>
   </widget>
   <widget class="QPushButton" name="atButton">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>110</y>
      <width>51</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>at</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="pushBackSpinBox">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>70</y>
      <width>121</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-2147483647</number>
    </property>
    <property name="maximum">
     <number>2147483647</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="pushFrontSpinBox">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>110</y>
      <width>121</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-2147483647</number>
    </property>
    <property name="maximum">
     <number>2147483647</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="atSpinBox">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>110</y>
      <width>61</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="atResLabel">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>110</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="isEmptyLabel">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>70</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>797</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
